vpdq/benchmark/benchmark.py

```diff
@@ -0,0 +1,151 @@
+        default=0,
+        type=int,
+    )
+    return ap
+
+
+def run_benchmark(
+    ffmpegPath: str,
+    secondsPerHash: int,
+    downsampleFrameDimension: int,
+):
+    with timer("Python hashing time") as pt:
+        python_features = run_python(
+            ffmpegPath, secondsPerHash, downsampleFrameDimension
+        )
+    feature_size = 0
+    for feature in python_features:
+        feature_size += len(feature)
+    python_time = pt()
+    print(
+        f"  Total hash:{ feature_size}  Per hash: {1000*python_time /  feature_size:.4f}ms"
+    )
+    with timer("CPP hashing time") as ct:
+        run_cpp(ffmpegPath, secondsPerHash, downsampleFrameDimension)
+    cpp_time = ct()
+    print(
+        f"  Total hash:{ feature_size}  Per hash: {1000*cpp_time /  feature_size:.4f}ms"
+    )
+    if downsampleFrameDimension != 0:
+        print(
+            "Calculate deviations in downsampled hashes, since hash resolution is non-native"
+        )
+        with timer("Original resolution Python hashing time"):
+            original_features = run_python(ffmpegPath, secondsPerHash, 0)
+        total_dist = 0
+        mis_match = 0
+        for original_hash, downsample_hash in zip(original_features, python_features):
+            for original_frame, downsample_frame in zip(original_hash, downsample_hash):
+                dist = original_frame.hamming_distance(downsample_frame)
+                if dist >= VPDQ_MATCH_DIST:
+                    mis_match += 1
+                total_dist += dist
+        print(
+            f"  Number of mismatches:{mis_match}, {mis_match/ feature_size:.2f} percent in total."
+        )
+        print(f"  Average {total_dist/ feature_size:.2f} hamming distance away.")
+
+
+def run_python(
+    ffmpegPath: str,
+    secondsPerHash: int,
+    downsampleFrameDimension: int,
+):
+    hash_list = []
+    for file in os.listdir(VIDEOS):
+        if file.endswith(".mp4"):
+            hash_list.append(
+                vpdq.computeHash(
+                    str(VIDEOS) + "/" + file,
+                    ffmpeg_path=ffmpegPath,
+                    seconds_per_hash=secondsPerHash,
+                    downsample_width=downsampleFrameDimension,
+                    downsample_height=downsampleFrameDimension,
+                )
+            )
+    return hash_list
+
+
+def run_cpp(
+    ffmpegPath: str,
+    secondsPerHash: int,
+    downsampleFrameDimension: int,
+):
+    for file in os.listdir(VIDEOS):
+        if file.endswith(".mp4"):
+            subprocess.call(
+                [
+                    str(CPP_EXEC),
+                    "-f",
+                    str(ffmpegPath),
+                    "-r",
+                    str(secondsPerHash),
+                    "-d",
+                    str(OUTPUT),
+                    "-s",
+                    str(downsampleFrameDimension),
+                    "-i",
+                    str(VIDEOS) + "/" + file,
```

Comment:
Nit: `str(VIDEOS / file)` is slightly safer since it will use with the equivalent of `os.path.join`